lib/cfa/matcher.rb

```diff
@@ -1,9 +1,32 @@
 module CFA
-  # Class used to create matcher, that allows to find specific option in augeas
-  # tree or subtree
-  # TODO: examples of usage
+  # The Matcher is used as a predicate on {AugeasElement}.
```

Comment:
I have even considered refactoring this to make it generic already, but YAGNI, so I'll wait until we actually have another parser. I'll add
> \# @note The coupling to AugeasTree, AugeasElement is not a goal. Once we have more parsers it will go away.